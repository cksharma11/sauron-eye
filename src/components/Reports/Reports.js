import React from "react";
import Report from "../Report/Report";

const Reports = ({ details, username }) => {
  return (
    <div className="details">
      {details.map(detail => {
        return <Report detail={detail} username={username} key={detail}/>;
      })}
    </div>
  );
};

export default Reports;
