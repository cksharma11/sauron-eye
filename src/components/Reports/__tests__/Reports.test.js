import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Reports from "../Reports";

configure({ adapter: new Adapter() });

describe("Reports", () => {
  it("should render properly", () => {
    const details = [
      {
        name: "test1",
        sha: "sha123",
        failedSuites: []
      },
      {
        name: "test1",
        sha: "sha123",
        failedSuites: []
      }
    ];
    const username = "test";
    const wrapper = shallow(<Reports details={details} username={username} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
