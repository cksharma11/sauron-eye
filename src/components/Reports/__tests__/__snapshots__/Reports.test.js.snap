// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports should render properly 1`] = `
"<div className=\\"details\\">
  <Report detail={{...}} username=\\"test\\" />
  <Report detail={{...}} username=\\"test\\" />
</div>"
`;
